File: 8. C/readability.c
#include <string.h>
#include <math.h>
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int calc_grade(int letters, int words, int sentences);

int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    int grade = calc_grade(letters, words, sentences);

    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

}

int count_letters(string text)
{
    int letters = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (islower(text[i]) || isupper(text[i]))
        {
            letters++;
        }
    }

    return letters;
}

int count_words(string text)
{
    int words = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }

    return (words + 1);
}

int count_sentences(string text)
{
    int sentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    return sentences;
}

int calc_grade(int letters, int words, int sentences)
{
    float L = (float)letters / (float)words * 100;
    float S = (float)sentences / (float)words * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8;

    return (int) round(index);
}

File: 13. Python/readability.py
from cs50 import get_string

text = get_string("Text: ")


def get_letters(text):
    letters = 0
    for i in text:
        if i.islower() or i.isupper():
            letters += 1
    return letters

def get_words(text):
    words = 0
    for i in text:
        if i == " ":
            words += 1
    return (words + 1)

def get_sentences(text):
    sentences = 0
    for i in text:
        if i == "." or i == "?" or i == "!":
            sentences += 1
    return sentences

def get_formula(letters, words, sentences):
    L = float(letters) / float(words) * 100.0
    S = float(sentences) / float(words) * 100.0
    formula = 0.0588 * L - 0.296 * S - 15.8
    return round(formula)

letters = get_letters(text)
words = get_words(text)
sentences = get_sentences(text)
grade = get_formula(letters, words, sentences)

if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {grade}")
